### PR TITLE
Add support for Material 3 `Divider` and `VerticalDivider`

### LINKED ISSUE #112169: Update `Divider`  to support Material 3
Specs: https://m3.material.io/components/divider/specs

image

We should update the `Divider` widget to support this Material 3 look and tokens.

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/112169

This PR adds support for Material 3 `Divider` and `VerticalDivider` and official M3 examples. Also cleaned up and added tests for existing divider examples.

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      themeMode: ThemeMode.light,
      theme: ThemeData(
        useMaterial3: true,
        brightness: Brightness.light,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      darkTheme: ThemeData(
        useMaterial3: true,
        brightness: Brightness.dark,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      home: const Example(),
    );
  }
}

class Example extends StatelessWidget {
  const Example({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Divider Sample'),
      ),
      body: Center(
        child: Row(
          children: const <Widget>[
            Expanded(child: Divider()),
            Expanded(child: VerticalDivider()),
          ],
        ),
      ),
    );
  }
}
``` 
	
</details>

| Preview | 
| --------------- | 
| 	|

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/divider/vertical_divider.1.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(title: const Text('Divder Sample')),
+        body: const DivierSample(),
+      ),
+    );
+  }
+}
+
+class DivierSample extends StatelessWidget {
```
Comment (darrenaustin) [inline examples/api/lib/material/divider/vertical_divider.1.dart:26]:
```suggestion
class DividerSample extends StatelessWidget {
```

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Thanks @TahaTesser. Looks good, just some typos that need to be fixed.

File: examples/api/lib/material/divider/vertical_divider.1.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(title: const Text('Divder Sample')),
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply examples/api/lib/material/divider/vertical_divider.1.dart:19]: Thanks!

File: examples/api/lib/material/divider/divider.1.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(title: const Text('Divder Sample')),
+        body: const DivierSample(),
+      ),
+    );
+  }
+}
+
+class DivierSample extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply examples/api/lib/material/divider/divider.1.dart:26]: Actually, it should be `DividerExample` (`Example`) to match the [example](https://github.com/flutter/flutter/blob/11f4d4fc02a1137f70bf8efc9c1d16d0d6547413/examples/api/lib/sample_templates/widgets.0.dart#L58) template and other existing examples.

**Review by darrenaustin — APPROVED** [review]
LTGM. Thx!

**Comment by kururu-abdo** [comment]
that's cool
so m3 now on flutter ?

**Comment by TahaTesser** [comment]
@kururu-abdo 
Please see https://github.com/flutter/flutter/issues/91605

**Pull request merged**